backend/api/serializers.py

```diff
@@ -224,10 +230,57 @@ def create(self, validated_data):
         user.save()
         return user
 
+    def update(self, instance, validated_data):
+        # username can't be changed
+        username = validated_data.get('username', None)
+        if username is not None and username != instance.username:
+            raise serializers.ValidationError('Username can\'t be changed.')
+
+        # password change
+        new_password = validated_data.get('new_password', None)
+        if new_password:
+            current_password = validated_data.get('current_password', None)
+            if check_password(current_password, instance.password):
+                instance.set_password(new_password)
+            else:
+                serializers.ValidationError('Current password is wrong.')
+
+        # yet email can be changed
+        instance.email = validated_data.get('email', instance.email)
```

Comment:
I think instead of explicitly setting all these fields, you can do something like

```
for key, value in validated_data.items():
    setattr(instance, key, value)
```

after you pop all the things from validated_data that requires manual handling. For example now, `profile_photo` isn't updated.